app/views/directives/create-secret.html

```diff
@@ -0,0 +1,307 @@
+      <div>
+        <input class="form-control"
+          id="dockerServer"
+          name="dockerServer"
+          ng-model="newSecret.data.dockerServer"
+          type="text"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+          required>
+      </div>
+      <div class="help-block">
+        Address of the registry that you will authenticate against.
+      </div>
+    </div>
+
+    <div class="form-group">
+      <label for="dockerUsername" class="required">Docker Username</label>
+      <div>
+        <input class="form-control"
+          id="dockerUsername"
+          name="dockerUsername"
+          ng-model="newSecret.data.dockerUsername"
+          type="text"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+          required>
+      </div>
+      <div class="help-block">
+        Username used for authenticate against secured registry.
+      </div>
+    </div>
+
+    <div class="form-group">
+      <label for="dockerPassword" class="required">Docker Password</label>
+      <div>
+        <input class="form-control"
+          id="dockerPassword"
+          name="dockerPassword"
+          ng-model="newSecret.data.dockerPassword"
+          type="password"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+          required>
+      </div>
+      <div class="help-block">
+        Password used for authenticate against secured registry.
+      </div>
+    </div>
+
+    <div class="form-group">
+      <label for="dockerEmail" class="required">Docker Email</label>
+      <div>
+        <input class="form-control"
+          type="email"
+          id="dockerEmail"
+          name="dockerEmail"
+          ng-model="newSecret.data.dockerMail"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+          required>
+      </div>
+      <div class="help-block">
+        Email used for authenticate against secured registry.
+      </div>
+    </div>
+  </div>
+
+  <div ng-if="newSecret.authType === 'Docker Config'">
+    <div class="form-group" id="docker-config">
+      <label for="dockerConfig" class="required">Docker Configuration File</label>
+      <osc-file-input 
+        model="newSecret.data.dockerConfig"
+        drop-zone-id="docker-config"
+        dragging="false"
+        help-text="Upload a .dockercfg or config.json file"
+        show-values="false"></osc-file-input>
+      <div ui-ace="{
+        mode: 'json',
+        theme: 'eclipse',
+        onLoad: aceLoaded,
+        rendererOptions: {
+          fadeFoldWidgets: true,
+          showPrintMargin: false 
+        }
+      }" ng-model="newSecret.data.dockerConfig" class="create-secret-editor ace-bordered" id="dockerconfig-editor" required></div>
+      <div class="help-block">
+        Content of Docker credentials file that stores information about registry you want to authenticate against.
```

Comment:
it takes both..
what about: "Docker configuration file with credentials and configuration to the secured registry." ?